app/models/comment.rb

```diff
@@ -327,13 +321,14 @@ def async_bust
   handle_asynchronously :async_bust
 
   def send_email_notification
-    NotifyMailer.new_reply_email(self).deliver if parent_email_exist?
+    NotifyMailer.new_reply_email(self).deliver
   end
   handle_asynchronously :send_email_notification
 
-  def parent_email_exist?
-    parent_user && parent_user.email.present? &&
-      parent_user.email_comment_notifications
+  def should_send_email_notification?
+    parent_user != user &&
```

Comment:
This makes sure article author/comment author won't receive an email notification if it is written by self.